[Security Solution] expandable flyout - folder structure

PhilippeOberti opened the pull request:
## Summary

This PR prepares the flyout folder for other teams to have their own expandable flyout in Security Solution:
- move most of the code into a new `document_details` folder under `flyout`
    - all the code specific to the document_details flyout was moved
    - a very few selected shared components is kept in a `shared` folder at the root of the `flyout` folder
- update paths accordingly
- update the README file for the `kbn/expandable-flyout` package
- update the README file for the Security Solution `flyout` folder
- added storybook for the `kbn/expandable-flyout` package

https://github.com/elastic/kibana/assets/17276605/244b51ad-1f96-4fb2-90f6-7623953a9c24



Absolutely no changes to UI or code logic.

https://github.com/elastic/security-team/issues/7626

PhilippeOberti committed:
[Security Solution] expandable flyout - folder structure + documentation

logeekal commented on packages/kbn-expandable-flyout/README.md:
```diff
@@ -9,30 +9,34 @@ The flyout is composed of 3 sections:
-The ExpandableFlyout [React component](https://github.com/elastic/kibana/tree/main/packages/kbn-expandable-flyout/src/components/index) that renders the UI.
+The expandable-flyout is making some strict UI design decisions:
+- when in collapsed mode (i.e. when only the right/preview section is open), the flyout's width is fixed to the EUI `s` size
+- when in expanded mode (i.e. when the left section is opened), the flyout's width is fixed to the EUI `l` size 
```
@PhilippeOberti , do you mean the whole flyout width or of a particular section. Also, how does expanded mode impacts the right section?

PhilippeOberti replied:
thanks I added a bit more precision. I didn't want to get too much into details as most developers wouldn't care but I think I found a good compromise

logeekal commented on packages/kbn-expandable-flyout/README.md:
```diff
@@ -9,30 +9,34 @@ The flyout is composed of 3 sections:
 - a preview section, that overlays the right section. This preview section can display multiple panels one after the other and displays a `Back` button
 
-At the moment, displaying more than one flyout within the same plugin might be complicated, unless there are in difference areas in the codebase and the contexts don't conflict with each other.
+## Design decisions
```
Hey @PhilippeOberti ,

1. I remember you were testing this component in Storybook so it will be great to mention storybook and if somebody wants to implement the flyout, how can they play with it in Storybook.
2. Secondly, it will be great to have some wireframe/screenshot here so that people can imagine what is left/right/preview panel.

PhilippeOberti replied:
great idea. I wasn't sure it'd be easy to setup Storybook for a package but it's exactly the same as a plugin. Do you think having Storybook would be enough, or addin g a screenshot would be useful here?


https://github.com/elastic/kibana/assets/17276605/3b5c0aca-6e6e-4f74-9ac8-358edb478083

logeekal replied:
This is great. I think just story book instructions should be fine as well. This will make adoption much easier.

logeekal reviewed (approved):
@PhilippeOberti , Thanks for taking time in putting together documentation. I think it is very clear. Although, I had couple of comments. Still I am approving the PR based on the READMEs.

christineweng commented on x-pack/plugins/security_solution/public/flyout/README.md:
```diff
@@ -1,44 +1,61 @@
-At the moment, we only have a single expandable flyout for the Security Solution plugin. This flyout will be used for all documents (signals, events, indicators, assets and findings). We're using a set of generic right/left/preview panels, hence the following folder structure:
-
+The structure of the `flyout` folder is intended to work as follows:
+- top level folders refer to the _type_ of flyout (for example alert details, event details, user, host, rule, cases...) and would contain all the panels for that flyout _type_.
```
should `alert details, event details` => `document details`?

PhilippeOberti replied:
thank you, fixed!

christineweng commented on x-pack/plugins/security_solution/public/flyout/README.md:
```diff
@@ -1,44 +1,61 @@
-│   └───right
-│   └───left
-│   └───preview
+└─── event_details
```
doesn't document details include event_details?

PhilippeOberti replied:
yup you are correct, removed

christineweng commented on x-pack/plugins/security_solution/public/flyout/README.md:
````diff
@@ -1,44 +1,61 @@
+    └─── components
 ```
+
+Where the top level `shared` folder contains all the components/hooks/services/helpers that are used accross multiple flyout types.
````
nit: consider moving this section before line 25. i think it helps to layout there are 2 types of top level folders: 1) type folders, 2) shared folder, then show suggested structure at the end. wdyt?

PhilippeOberti replied:
I moved this section up and rewrote the bullet points, let me know if they make more sense

PhilippeOberti committed:
PR comments:
- adding storybook to kbn-expandable-flyout package
- improve kbn-expandable-flyout package README
- add _TEST_ID to kbn-expandable-flyout package test ids

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

PhilippeOberti committed:
some more PR comments:
- improve Security Solution flyout README
- added stories for all shared components
